backend/src/apiserver/config/proxy/config.go

```diff
@@ -0,0 +1,132 @@
+// Copyright 2025 The Kubeflow Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package proxy
+
+import (
+	k8score "k8s.io/api/core/v1"
+	"os"
+)
+
+const (
+	HttpProxyEnv        = "HTTP_PROXY"
+	HttpsProxyEnv       = "HTTPS_PROXY"
+	NoProxyEnv          = "NO_PROXY"
+	defaultNoProxyValue = "localhost,127.0.0.1,.svc.cluster.local,kubernetes.default.svc,metadata-grpc-service,0,1,2,3,4,5,6,7,8,9"
+)
+
+type Config interface {
+	GetHttpProxy() string
+	GetHttpsProxy() string
+	GetNoProxy() string
+	GetEnvVars() []k8score.EnvVar
+}
+
+var (
+	configInstance Config = &nonInitializedConfig{}
+	emptyConfig           = newConfig("", "", "")
```

Comment:
I would have expected empty config to have the default no proxy set but I see that it's only used in the tests.

Reply:
Maybe we can rename it.
By "Empty", one can think of `newConfig("", "", "")`, which is a valid value (and not really empty), or the default values.
How about `blankConfig`? As a non-native English speaker, I'm not sure how different from `empty` it really is.

Reply:
I think `empty` is fine. I think I was equating `empty` to `default` but it's not the same.